core/src/main/java/org/phantazm/core/player/BasicPlayerView.java

```diff
@@ -126,6 +126,17 @@ private CompletableFuture<String> getUsernameRequest() {
         return getUsername().thenApply(Component::text);
     }
 
+    @Override
+    public @NotNull Component getDisplayNameIfPresent() {
+        Optional<? extends Component> cached = getPlayer().map(Player::getDisplayName);
+        if (cached.isPresent()) {
+            return cached.get();
+        }
+
+        return getUsernameIfCached().map(Component::text).orElseGet(() -> Component.text(getUUID().toString()));
+
+    }
+
```

Comment:
very similar to the method right below (which I can't highlight)

Reply:
difference is it won't make API calls to mojang, it's basically getDisplayNameIfCached() (should probably call it that)

Reply:
well the other is an Optional which is a bit messier to work with

Reply:
w/e its not that important